File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import dto.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌服务层接口
 *
 * @Author ll
 * @Date 2018/10/30 11:18
 */
public interface BrandService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 查询分页对象
     *
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 添加品牌
     *
     * @param brand
     */
    public void add(TbBrand brand);

    /**
     * 查询品牌
     *
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 修改品牌
     *
     * @param brand
     */
    public void update(TbBrand brand);

    /**
     * 删除品牌
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 查询品牌分页
     *
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbBrand brand, int pageNum, int pageSize);

    /**
     * 获取所有品牌信息
     *
     * @return
     */
    public List<Map> findBrandList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?num=' + page + '&size=' + rows);
    };
    this.search = function (page, rows, searchEntity) {
        return $http.post('../brand/search.do?num=' + page + '&size=' + rows, searchEntity);
    };
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    };
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    };
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    };
    this.delete = function (selectIds) {
        return $http.get("../brand/delete.do?ids=" + selectIds);
    };
});